sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/export/RegisteredReaderTest.java

```diff
@@ -20,8 +21,8 @@ class RegisteredReaderTest {
 
   @Test
   void create_UniqueIdentity() {
-    RegisteredReader registeredReader1 = RegisteredReader.create(reader);
-    RegisteredReader registeredReader2 = RegisteredReader.create(reader);
+    RegisteredReader registeredReader1 = RegisteredReader.create(reader, ViewRegistry.create());
```

Comment:
Does this mean each reader gets its own set of views?

It seems like the "ViewRegistry"is actually the new proxy for "in-memory storage" of metrics with this PR.  I'm not sure if a different name would help here (or description).

Reply:
> Does this mean each reader gets its own set of views?

Yes. This helps accommodate the differences that the SDK spec already allows for from reader to reader, like each being able to dictate its own aggregation temporality and (as of this PR) its own default aggregation. But it also sets us up neatly for future (currently proposed) spec changes where each reader could have its own set of views. One real world application of this would be to have different collection intervals for different instruments - maybe you want 5s granularity of JVM runtime metrics like memory usage, but 60s granularity is enough for other instruments. If each reader has its own view configuration you register two periodic metric readers. The first has a 5s interval and uses views to drop all metrics except JVM runtime. The second has a 60s interval and uses views to drop all JVM runtime metrics.

> It seems like the "ViewRegistry"is actually the new proxy for "in-memory storage" of metrics with this PR. I'm not sure if a different name would help here (or description).

The key API of view registry is `List<RegisteredView> findViews(InstrumentDescriptor, InstrumentationScopeInfo)`. This is called once per registered reader when an instrument is created. Each `RegisteredView` in the response is used to create a `MetricStorage`. Each measurement recorded from an instrument is sent to all the associated `MetricStorages`. So effectively each instrument results metric storages equal to `num readers * num views`. 

I think `ViewRegistry` is still the correct name, as it reflects its core function of defining which views should be created for each instrument. Those views ultimately result in metric storages being created, but that's not a concern of `ViewRegistry`.